userspace/libgolf2/aes_ecb_syscalls.h

```diff
@@ -0,0 +1,114 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "tock.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// Checks whether there is an AES system call driver. Returns TOCK_SUCCESS
+// if there is, ENOSUPPORT otherwise.
+int tock_aes128_check(void);
+
+// Configures the encryption key to be used for encryption and decryption.
+//
+// key - a buffer containing the key (must be 16 bytes)
+// len - length of the buffer (must be 16)
+int tock_aes128_set_key(unsigned char* key, unsigned char len);
+
+
+// Encrypts a payload according to aes-128 counter-mode. The counter
+// stored in ctr is incremented for each block encrypted in a single call.
+// The caller is responsible for incrementing/managing the counter across
```

Comment:
"The caller is responsible for incrementing/managing the counter across calls." --> This implies that the user must increment ctr between calls to tock_aes128_encrypt_ctr_sync(), but that's not true (based on the rest of the comment).

I think the sentence can be removed without hurting understanding.

Reply:
What that meant is that you'll need to pass the counter back; I agree, though, that it's redundant with the rest of the text I added. Removed.